filebeat/harvester/reader/docker_json.go

```diff
@@ -39,27 +95,19 @@ func (p *DockerJSON) Next() (Message, error) {
 			return message, err
 		}
 
-		var line dockerLog
-		dec := json.NewDecoder(bytes.NewReader(message.Content))
-		if err = dec.Decode(&line); err != nil {
-			return message, errors.Wrap(err, "decoding docker JSON")
-		}
+		var dockerLine dockerLog
+		var crioLine crioLog
 
-		if p.stream != "all" && p.stream != line.Stream {
-			continue
+		if strings.HasPrefix(string(message.Content), "{") {
+			message, err = parseDockerJSONLog(message.Content, &dockerLine, message)
```

Comment:
Having that you are passing `message` anyway, you could reduce the number of params to 2 (`&message` and `&dockerLine`). Same applies to parseCRILog

Reply:
Agree! I'll reduce it to 2 params.